### PR TITLE
metrics(*): fix for promtheus_client breaking change

### LINKED ISSUE #5494: Revert prometheus client_golang from  from v1.12.2 to v1.11.0
### What did you do?

When merge engine to tiflow repo, client_golang(v1.12.2) will introduce some incompatible change to the go collector and some metric name.

**Detail**
ref : https://github.com/prometheus/client_golang/releases
[CHANGE] ⚠️ Change in collectors.NewGoCollector metrics: Reverting addition of new ~80 runtime metrics by default. You can enable this back with GoRuntimeMetricsCollection option or GoRuntimeMemStatsCollection | GoRuntimeMetricsCollection for smooth transition.
[BUGFIX] Fixed the bug that causes generated histogram metric names to end with _total. ⚠️ This changes 3 metric names in the new Go collector that was reverted from default in this release.
go_gc_heap_allocs_by_size_bytes_total -> go_gc_heap_allocs_by_size_bytes,
go_gc_heap_frees_by_size_bytes_total -> go_gc_heap_allocs_by_size_bytes
go_gc_pauses_seconds_total -> go_gc_pauses_seconds.

### What did you expect to see?

_No response_

### What did you see instead?

some go collector metric lost

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
master
```

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

Issue Number: close #5494

### What is changed and how it works?

add flag for smooth transition, see more in  https://github.com/prometheus/client_golang/releases/tag/v1.12.2

and https://github.com/pingcap/tiflow/pull/5393#discussion_r884428789
#### Tests 

the breaking changed metrics are not used in our code

* go_gc_heap_allocs_by_size_bytes_total -> go_gc_heap_allocs_by_size_bytes,
* go_gc_heap_frees_by_size_bytes_total -> go_gc_heap_allocs_by_size_bytes
* go_gc_pauses_seconds_total -> go_gc_pauses_seconds.

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by Ehco1996** [comment]
ptal

/cc @maxshuang  @overvenus

File: dm/dm/worker/metrics.go [hunk 1]
```diff
@@ -104,7 +105,8 @@ func (s *Server) runBackgroundJob(ctx context.Context) {
 func RegistryMetrics() {
 	registry := prometheus.NewRegistry()
 	registry.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
-	registry.MustRegister(prometheus.NewGoCollector())
+	registry.MustRegister(prometheus.NewGoCollector(
```
Comment (maxshuang) [inline dm/dm/worker/metrics.go:108]: Pls modify the   `NewGoCollector()` in  cdc/metrics.go.

Comment (Ehco1996) [reply dm/dm/worker/metrics.go:108]: thanks addressed in https://github.com/pingcap/tiflow/pull/5643/commits/c6ac8c0155eeb449266a354f09006a430a87ae70

**Comment by maxshuang** [comment]
Rest LGTM.

**Review by maxshuang — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**